pennylane/beta/tapes/qnode.py

```diff
@@ -207,7 +207,7 @@ def get_best_method(device, interface):
                 return QNode._validate_device_method(device, interface)
             except QuantumFunctionError:
                 try:
-                    return QNode._get_parameter_shift_method(device, interface)
+                    return QNode._get_parameter_shift_tape(device), interface, "best"

```

Comment:
This ensures that if the user gets the parameter-shift rule automatically (by specifying `diff_method="best"`) that the jacobian will support finite-difference fallback for any non-analytic gate argument.